### PR TITLE
Update schemer to use current versions of crate dependencies.

### PR DESCRIPTION
**Pull request by nuttycom**
Hi there! Thanks a lot for this library; you handle migrations exactly as I think they should be done. I'd like to revive this library, so this PR does a little bit of modernization. Also, I'd be happy to contribute to future maintenance.

### CODE REVIEW

**Comment by aschampion** [comment]
Thanks for the contributions! I'll have a look this weekend. There are some overlapping changes in an old `renovation` branch, so I'll merge those and also hook up the CI workflows I've started using in my other crates.

File: schemer/src/lib.rs [hunk 1]
```diff
@@ -142,11 +137,11 @@ pub enum DependencyError {
     DuplicateId(Uuid),
     #[fail(display = "Unknown migration ID {}", _0)]
     UnknownId(Uuid),
-    #[fail(display = "Cyclic dependency cased by edge from migration IDs {} to {}", from, to)]
-    Cycle {
-        from: Uuid,
-        to: Uuid,
-    }
+    #[fail(
+        display = "Cyclic dependency cased by edge from migration IDs {} to {}",
```
Comment (daira) [inline schemer/src/lib.rs]:
```suggestion
        display = "Cyclic dependency caused by edge from migration IDs {} to {}",
```

**Review by daira — APPROVED** [review]
Untested ACK

**Comment by aschampion** [comment]
I found a moment to merge the old cleanup, set up basic CI again, and do minimal dependency updates for some downstream private dependencies I have. If you could rebase I'll merge, or otherwise confirm that it looks like the remaining dep updates you need are `daggy` and `rusqlite`.

The old cleanup also changed from `failure` to `thiserror` (as the rest of my crates have). I assume this isn't a problem for you downstream.

**Comment by str4d** [comment]
> The old cleanup also changed from `failure` to `thiserror` (as the rest of my crates have). I assume this isn't a problem for you downstream.

Nope, no problem for us :)

**Comment by nuttycom** [comment]
@aschampion thanks for all these updates. I've updated both this and #14 to be based on the current `master` branch.

**Pull request merged**